modules/storage/storage-account/main.bicep

```
@@ -19,6 +19,142 @@ param isZoneRedundant bool = false
 @description('Storage Account Type. Use Zonal Redundant Storage when able.')
 param storageAccountType string = isZoneRedundant ? 'Standard_ZRS' : 'Standard_LRS'
 
+// @description('Toggle to enable or disable Blob service of the Storage Account.')
+// param enableBlobService bool = false
+
+@description('Name of a blob service to be created.')
+param blobName string = 'default'
+
+@description('Properties object for a Blob service of a Storage Account.')
+param blobProperties blobServiceProperties = {}
+
+@description('Name of a blob container to be created')
+param blobContainerName string = 'default'
+
+@description('Properties object for a Blob container of a Storage Account.')
+param blobContainerProperties blobServiceContainerProperties = {}
+
+@description('Array of role assignment objects that contain the \'roleDefinitionIdOrName\', \'principalId\' and \'resourceType\' as \'storageAccount\' or \'blobContainer\' to define RBAC role assignments on that resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'')
+param roleAssignments roleAssignmentsArray = []
+
+type blobServiceProperties = {
+  changeFeed: changeFeed?
+  containerDeleteRetentionPolicy: containerDeleteRetentionPolicy?
+  cors: cors?
+  deleteRetentionPolicy: deleteRetentionPolicy?
+  isVersioningEnabled: isBlobVersioningEnabled?
+  lastAccessTimeTrackingPolicy: lastAccessTimeTrackingPolicy?
+  restorePolicy: restorePolicy?
+}
+
+type changeFeed = {
+  enabled: bool
+  retentionInDays: changeFeedretentionInDays?
+}
+
+@minValue(1)
+@maxValue(146000)
+type changeFeedretentionInDays = int
+
+type containerDeleteRetentionPolicy = {
+  allowPermanentDelete: bool
+  days: containerDeleteRetensionPolicyDays?
+  enabled: bool
+}
+
+@minValue(1)
+@maxValue(365)
+type containerDeleteRetensionPolicyDays = int
+
+type cors = {
+  corsRules: [
+    {
+      allowedHeaders: [
+        'string'
+      ]
+      allowedMethods: [
+        'string'
+      ]
+      allowedOrigins: [
+        'string'
+      ]
+      exposedHeaders: [
+        'string'
+      ]
+      maxAgeInSeconds: int
+    }
+  ]
+}
+
+type deleteRetentionPolicy = {
+  allowPermanentDelete: bool
+  days: deleteRetentionPolicyDays?
+  enabled: bool
+}
+
+@minValue(1)
+@maxValue(365)
+type deleteRetentionPolicyDays = int
+
+@description('Toggle to enable or disable versioning for Blob service of the Storage Account. Used only if enableBlobService is set to true.')
+type isBlobVersioningEnabled = bool
+
+type lastAccessTimeTrackingPolicy = {
+  blobType: [
+    'string'
+  ]
+  enable: bool
+  name: 'AccessTimeTracking'
+  trackingGranularityInDays: trackingGranularityInDays?
+}
```

```suggestion
type lastAccessTimeTrackingPolicy = {
  blobType: [
    'string'
  ]
  enable: bool
  name: 'AccessTimeTracking'
  trackingGranularityInDays: trackingGranularityInDays?
}
```

blobType, name, and trackingGranularityInDays are read only. Do we need to include them.